### PR TITLE
Feat/user orders history customer

### PR DESCRIPTION
**Pull request by rolandiartmeladze**
# Pull Request Template for ArtOa

## Description

This pull request introduces features to display the user orders history, categorized by different order statuses. The data is organized into groups like "Current", "Finished", and "Proposed", and displayed via a carousel UI component.

#### What does this PR do?
- Implements a grouped order history display based on the order's status ("Current", "Finished", "Proposed").
- Uses the OrderCard component to display individual orders in a carousel layout.
- 
#### Why are these changes necessary?
- This update is necessary to allow customers to easily view their order history, sorted by status. The carousel component ensures that orders are displayed efficiently and in a visually appealing manner.
- 
#### Any relevant links or references?

Fixes: [#78]

---

## Type of Change

Please select the type of change this PR introduces (tick the appropriate box):

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- [ ] 🔄 Refactor (code or file structure changes without affecting functionality)
- [ ] 📝 Documentation update (changes or additions to documentation)
- [ ] 🚀 Performance improvement (optimization in code or resources)

---

## Checklist

Please ensure the following tasks have been completed before requesting a review:

- [x] My code follows the style guidelines of this project
- [ ] I have run tests to ensure my changes work as expected
- [ ] I have added/updated necessary tests or documentation (if applicable)
- [ ] I have self-reviewed my code and refactored any areas that need improvement
- [ ] There are no console errors or warnings related to my changes
- [ ] The PR includes a clear, detailed title and description
- [ ] Any relevant issues have been linked or mentioned in this PR
- [ ] I have reviewed the acceptance criteria and met all requirements
- [ ] I have considered edge cases and errors that may occur due to my changes

---

## Testing Instructions

Please describe how your changes can be tested:

1. **Test Setup:** (Any steps for environment setup)
2. **Test Steps:**
   
- Step 1: Verify that the orders are correctly grouped by status ("Current", "Finished", "Proposed").
-  Step 2: Ensure that each order is displayed using the OrderCard component within the carousel.
- Step 3: Verify that the carousel works as expected on different screen sizes.
   - ...

---

## Screenshots / Visual Changes

https://github.com/user-attachments/assets/2eb10a08-8a66-4c31-8e06-3f6c584afd45

orders1
orders2
orders3

Note: This is the current visual state and is still in progress. Further updates and improvements are underway.

---

## Additional Notes

Is there anything else the reviewer should know about this PR? Any known issues,
challenges, or decisions made during implementation?

### CODE REVIEW

File: src/styles/globals.css [hunk 1]
```diff
@@ -16,6 +16,11 @@
   -ms-overflow-style: none;
 }
 
+[data-radix-popper-content-wrapper] {
+  position: absolute !important;
```
Comment (chubinaaa) [inline src/styles/globals.css:20]: მომავლისთვის ეცადე რომ !important-ის გამოყენება არ დაგჭირდეს. ახლა იყოს როგორც არის

File: src/config/orders-history.tsx
Comment (chubinaaa) [inline src/config/orders-history.tsx:1]: ფაილს `.tsx` გაფართოება აქვს და უბრალოდ ჯავასკრიპტის ობიექტს აბრუნებს. შემდგომში დააკვირდი მსგავს დეტალებს.

File: src/components/ui/popover.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+const Popover = PopoverPrimitive.Root;
+
+const PopoverTrigger = PopoverPrimitive.Trigger;
+
+const PopoverAnchor = PopoverPrimitive.Anchor;
+
+const PopoverContent = React.forwardRef<
+  React.ElementRef<typeof PopoverPrimitive.Content>,
+  React.ComponentPropsWithoutRef<typeof PopoverPrimitive.Content>
+>(({ className, align = "center", sideOffset = 4, ...props }, ref) => (
```
Comment (chubinaaa) [inline src/components/ui/popover.tsx:17]: აქვე დეფოლტ ველიუს მინიჭება ტექნიკურად კი შესაძლებელია მაგრამ აჯობებს სხვაგან მიანიჭო ან სხვანაირად დაწერო

File: src/app/(user)/(profile)/profile/components/order-popover.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+export default function OrderPopover({ order, open, onClose }: OrderPopover) {
+  return (
+    <Popover open={open} onOpenChange={onClose}>
+      <PopoverContent className="absolute top-60 flex w-full flex-col items-center gap-4 rounded-t-xl bg-background p-4">
+        <div className="h-1 w-12 rounded-full bg-foreground"></div>
+        <div className="flex flex-col items-center gap-4">
+          <div className="flex w-full justify-center gap-4 border-b border-secondary-foreground pb-4">
+            <samp className="inline rounded-md border border-secondary-foreground p-2 text-sm font-semibold text-foreground">
```
Comment (chubinaaa) [inline src/app/(user)/(profile)/profile/components/order-popover.tsx:19]: მსგავსი ტეგების გამოყენებაც არ არის საჭირო. ახალებია და სჯობს ნაცადი p და heading ტეგები გადავსტილოთ ისე როგორც საჭიროა

File: src/app/(user)/(profile)/profile/components/order-card.tsx
Comment (chubinaaa) [inline src/app/(user)/(profile)/profile/components/order-card.tsx:1]: შენს ადგილას ქარდ ჰედერს ცალკე წავიღებდი, ქარდ ბადის ცალკე და ეს კომპონენტი მაგით გამარტივდებოდა, ერთმანეთთან მაინც არაფერი აქვთ საერთო. სტეიტს კი გაიზიარებდნენ

გახსოვდეს, 'დაყავი და იბატონე' - რომაელების გარდა რეაქტ დეველოპერებსაც გვადგება.

File: src/app/(user)/(profile)/profile/components/grouped-orders.tsx
Comment (chubinaaa) [inline src/app/(user)/(profile)/profile/components/grouped-orders.tsx:1]: აქაც მსგავსად

ერთ კომპონენტში ორი გადამაპვა წერია და დამღლელია თვალისთვის უბრალოდ, სულ რომ არაფერი. შემდგომი maintenance-ისთვისაც

**Review by chubinaaa — APPROVED** [review]
